### ISSUE TITLE
Unsupported Recurrence

### ISSUE THREAD

**Issue by Petemir** [pimutils/khal#1020]
Hello,

I am just starting with khal, I wanted first to thank all maintainers for their effort in supporting this interesting type of software :) .

That said, I just synced with vsyncdir some Google calendars and I got some errors when trying to start khal (installed via pip, so I have the latest release). Some information has been redacted (cal name, event id, paths, and event name).

I'll add the details as per [here](https://khal.readthedocs.io/en/latest/feedback.html), if there's anything else that you may need let me know. Thanks!

* how you ran khal (please run in verbose mode with -v DEBUG):
"khal", "khal list" , "khal interactive"

This doesn't happen always, so I did it first, found the error, and then trying to run it with -v DEBUG didn't return the error... but then I just did a normal "khal interactive" again and the error appeared again :man_shrugging: .

* what you expected khal to do:
To start without any warnings nor errors

* what it did instead:
It printed an error

* everything khal printed to the screen (you may redact private details):
```
warning: Skipping $CALNAME/$EVENTID.ics: 
warning: This event will not be available in khal.
Unknown exception happened.
Traceback (most recent call last):
  File "~/anaconda3/envs/newbase/lib/python3.8/site-packagesy", line 338, in _update_vevent
    update(event.raw, href=href, etag=etag, calendar=calendar)
  File "~/anaconda3/envs/newbase/lib/python3.8/site-packages, line 240, in update
    self._update_impl(vevent, href, calendar)
  File "~/anaconda3/envs/newbase/lib/python3.8/site-packages, line 356, in _update_impl
    dtstartend = expand_vevent(vevent, href)
  File "~/anaconda3/envs/newbase/lib/python3.8/site-packages72, in expand
    raise UnsupportedRecurrence()
khal.exceptions.UnsupportedRecurrence
```
* in case khal complains about a specific .ics file, please include that as well (or create a .ics which leads to the same error without any private information)

```
BEGIN:VCALENDAR
PRODID:-//Google Inc//Google Calendar 70.9054//EN
VERSION:2.0
CALSCALE:GREGORIAN
X-WR-CALNAME:$CALNAME
X-WR-TIMEZONE:Europe/Zurich
BEGIN:VTIMEZONE
TZID:America/Argentina/Buenos_Aires
X-LIC-LOCATION:America/Argentina/Buenos_Aires
BEGIN:STANDARD
TZOFFSETFROM:-0300
TZOFFSETTO:-0300
TZNAME:-03
DTSTART:19700101T000000
END:STANDARD
END:VTIMEZONE
BEGIN:VEVENT
DTSTART;TZID=America/Argentina/Buenos_Aires:20190103T094500
DTEND;TZID=America/Argentina/Buenos_Aires:20190103T103000
RRULE:FREQ=WEEKLY;WKST=SU;UNTIL=20190108T025959Z;BYDAY=TU
DTSTAMP:20181227T185734Z
UID:$EVENTUID
CREATED:20181227T185644Z
DESCRIPTION:
LAST-MODIFIED:20181227T185734Z
LOCATION:$EVENTLOCATION
SEQUENCE:3
STATUS:CONFIRMED
SUMMARY:$EVENTNAME
TRANSP:OPAQUE
BEGIN:VALARM
ACTION:DISPLAY
DESCRIPTION:This is an event reminder
TRIGGER:-P0DT1H15M0S
END:VALARM
BEGIN:VALARM
ACTION:DISPLAY
DESCRIPTION:This is an event reminder
TRIGGER:-P0DT0H10M0S
END:VALARM
BEGIN:VALARM
ACTION:DISPLAY
DESCRIPTION:This is an event reminder
TRIGGER:-P0DT0H45M0S
END:VALARM
END:VEVENT
END:VCALENDAR
```
* the version of khal and python you are using, which operating system you are using and how you installed khal
khal: 0.10.2
python: 3.8.5
os: Ubuntu 20.10 with i3 and Plasma

**Comment by d7415** [comment]
The issue appears to be that it recurs every Tuesday at 0945, between 2019-01-03 (a Thursday) until 2019-01-08 at 02:59:59 (a Tuesday, but before 0945). If the UNTIL is changed to 2019-01-09 then it imports successfully.

As usual, the issue is "this event makes no sense". We could work around it, but only if we can figure out a sensible interpretation of it.

**Comment by Petemir** [comment]
@d7415 Thank you for your quick answer. As it is an event from 2 years ago, I have no recollection how it could have reached that inconsistent? state.  It was a recurring (weekly) event that changed times several times, so perhaps that's why one "UNTIL" finishes "early"? (i.e., I modified it in Google as "modify all future events"). I also changed timezones (from GMT-3 to GMT+1), but I don't think that action (made on 2019-09) could have an impact on previous events.
